src/python/turicreate/test/test_object_detector.py

```diff
@@ -288,6 +288,21 @@ def test_create_with_verbose_False(self):
         kwargs = {"max_iterations": 1, "model": self.pre_trained_model}
         test_util.assert_longer_verbose_logs(tc.object_detector.create, args, kwargs)
 
+    def test_create_with_fixed_random_seed(self):
+         random_seed = 86
+         max_iterations = 3
+
+         model_1 = tc.object_detector.create(self.sf, max_iterations=max_iterations, random_seed=random_seed)
+         pred_1 = model_1.predict(self.sf)
+         model_2 = tc.object_detector.create(self.sf, max_iterations=max_iterations, random_seed=random_seed)
+         pred_2 = model_2.predict(self.sf)
+
+         self.assertEqual(len(pred_1), len(pred_2))
+         for i in range(len(pred_1)):
+             self.assertEqual(len(pred_1[i]),len(pred_2[i]))
+             for j in range(len(pred_1[i])):
+                 self.assertEqual(pred_1[i][j], pred_2[i][j])
+
```

Comment:
I think this test is good, but it's really only testing that the same seed produces the same model with your local changes. It's not testing that both master and your local changes produce the same model with the same seed.

I suggest we hard code the probabilities for one row of the data here and check that those match.

Reply:
I don't think we're in a position to guarantee that the same golden output will remain valid. We already have to accept some numerical differences across platforms, and we can't control minor behavior changes in our dependencies across their versions. For example, we inherit whatever the semantics are for how numpy uses random seeding.

I do think we should revisit this test in general, since it's quite heavyweight for what it does. I filed #3094